### PR TITLE
Do not re-open deployment flow after initial commit

### LINKED ISSUE #1567: Deployment flow doesn't close on deploy with gisf/jem
Once you go through the deployment flow with gist and jem the deployment flow component closes then immediately re-opens. This appears to be a state race issue as it is being set to close then is set again to re-open.

### PR DESCRIPTION
**Pull request by hatched**
On the first deploy to a new model in the deployment flow the onLogin method is called because the user has logged into a new model. This preserves the old URL because of legacy login functionality. This fix disables that functionality when under gisf mode. Fixes #1567

### CODE REVIEW

**Comment by hatched** [comment]
#### To QA

In a real deployed GISF setup deploy a service, creating a model at the same time. It should closes the deploy summary after clicking Deploy.

File: jujugui/static/gui/src/app/app.js [hunk 1]
```diff
@@ -1869,16 +1868,13 @@ YUI.add('juju-gui', function(Y) {
...
         }
-        this.navigate(redirectPath, {overrideAllNamespaces: true});
-        // If the redirectPath has a hash then it will not dispatch after log in
-        // because navigateOnHash is set to false so that we can use hash's to
-        // show the correct tab in the charm details pages. This issue only
-        // presents itself until the next delta comes in and the application
-        // hits it's double dispatch and dispatches the url again. As a
-        // workaround we check if there is a hash in the url and then dispatch
-        // manually.
-        if (redirectPath.indexOf('#') > -1) {
```
Comment (hatched) [inline jujugui/static/gui/src/app/app.js]: We no longer have this limitation in the GUI because we no longer use any hash based routing so I just removed it as a driveby.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4900/
Test PASSed.

**Comment by huwshimi** [comment]
QA OK in Juju 1.5 :+1:

**Comment by urosj** [comment]
👍

**Comment by urosj** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge
